library/coretests/tests/num/mod.rs

```diff
@@ -924,6 +924,77 @@ macro_rules! test_float {
                 assert!($inf.div_euclid($nan).is_nan());
                 assert!($nan.div_euclid($inf).is_nan());
             }
+            #[test]
+            fn floor() {
+                $fassert!((0.0 as $fty).floor(), 0.0);
+                $fassert!((-0.0 as $fty).floor(), -0.0);
```

Comment:
Note that `0.0 == -0.0`, so you are not actually testing that the zero has the right sign here.

Maybe the `float_assert` and `float_const_assert` macros should compare `$left.to_bits()` and `$right.to_bits()` instead?

Reply:
That's a good point. I tried doing that, but it caused many compilation errors because e.g. `0.0` is of unknown float type. A solution for that would be to write `0.0 as $fty` in this entire file (including the other existing tests), which, in turn, caused some tests to fail (I think because of under- or overflows).

To avoid having this grow too much in scope, I followed the other tests' strategy by adding `.is_sign_positive()` and `.is_sign_negative()` tests. They allow us to check that the zero has the correct sign.

Please let me know if this strategy is OK, or if you'd still prefer the `.to_bits()` solution.